### PR TITLE
Render YAML frontmatter as a quiet disclosure

### LINKED ISSUE #43: Render YAML frontmatter without crowding the article
## Description

Markdown files with YAML frontmatter (skills, agent definitions, prompts, etc.) currently render the `---`-fenced block as a full-width 2-column key/value table at the top of the document. The table is visually heavier than the article H1 that follows it, gives equal weight to short fields like `name` and 50-word `description` paragraphs, and treats every key the same way regardless of how relevant it is to the reader.

## Why

Frontmatter is metadata, not content. For most readers it's noise above the actual document; for the readers who do care about it, a compact summary is enough until they want detail. The current rendering pushes the article body well below the fold and clashes with the app's quiet, paper/ochre tone.

## Acceptance Criteria

- [ ] Documents with YAML frontmatter no longer lead with a heavy full-width metadata table.
- [ ] Frontmatter remains accessible on the rendered page; readers who want it can see all fields.
- [ ] The article H1 and opening prose are visible above the fold on a typical viewport for documents that have frontmatter.
- [ ] Rendering works for arbitrary frontmatter shapes — not just skill files.

### PR DESCRIPTION
**Pull request by niftymonkey**
Closes #43.

## Summary

- New `parseFrontmatter` util extracts the leading `---`-fenced block from a markdown document; returns the parsed key/value pairs plus the body with the block stripped.
- New `<FrontmatterDisclosure>` server component renders a native `<details>` above the article: collapsed by default with a one-line preview (`name: <value>` + `+N fields` count), expands to a `dl` of all keys/values.
- The reader page strips frontmatter before parsing headings and before passing content to `MarkdownRenderer`, so the outline and the body render against just the prose.
- Sticky tops on the outline aside (`top-12`) and reader toolbar cluster (`top-14`) brought down to `top-10` so the side chrome's top edge aligns with the article's first-content top edge — true for both H1-leading docs and frontmatter-leading docs.

## Notes

- Schema-agnostic: the parser handles `key: value` lines (including indented continuation lines) and arbitrary key names. It doesn't try to handle nested objects/arrays or anchors — vanishingly rare in skill/agent/prompt files, and the worst case is a verbatim string in the rendered disclosure.
- No client JS needed for open/close — it's a native `<details>` element. Safe in the SSR'd reader.
- Unknown extras just appear as additional `dt`/`dd` rows; nothing about the rendering is coupled to a specific frontmatter shape.

11 new unit tests cover empty frontmatter, missing closing fence, CRLF endings, quoted values, continuation lines, and the body-thematic-break edge case.

## Summary by CodeRabbit

* **New Features**
  * Added a collapsible frontmatter section that displays article metadata as expandable key-value pairs.

* **Style**
  * Added styling for the frontmatter disclosure section with custom chevron animations and definition list layout.
  * Adjusted sidebar outline spacing and positioning.

* **Tests**
  * Added comprehensive test coverage for frontmatter parsing functionality.

### CODE REVIEW

File: src/components/frontmatter-disclosure.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        )}
+      </summary>
+      <div className="frontmatter__body">
+        <dl className="frontmatter__dl">
+          {fields.map((field) => (
+            <Fragment key={field.key}>
+              <dt className="frontmatter__dt">{field.key}</dt>
+              <dd className="frontmatter__dd">{field.value}</dd>
+            </Fragment>
+          ))}
```
[the remark this answers is not in the document]
Comment (niftymonkey) [reply src/components/frontmatter-disclosure.tsx]: Fixed in 770d665 — switched to `${field.key}-${index}` for the React key so duplicate frontmatter keys don't collide.
